zerver/views/messages.py

```diff
@@ -319,7 +340,9 @@ def highlight_string_text_offsets(text, locs):
     pos = 0
     result = u''
     for loc in locs:
-        (offset, length) = loc
+        (offset, length) = highlight_match_tag_pos(string, pos, loc, u'<', u'>')
+        if offset < pos:
+            continue
```

Comment:
Well, all pgroonga emoji search tests should definitely run into this case, since otherwise it would start repeating part of the tag (you might just as well test after removing it, to see that it fails).